libs/wingsdk/src/std/shared-state.ts

```diff
@@ -0,0 +1,69 @@
+import { Construct } from "constructs";
+import { IResource, Resource } from "./resource";
+import { fqnForType } from "../constants";
+import { App } from "../core";
+
+/**
+ * Global identifier for `SharedState`.
+ */
+export const SHARED_STATE_FQN = fqnForType("std.SharedState");
+
+/**
+ * A container for a value that can be shared and mutated across inflight consumers.
+ * Only supported in the `sim` compilation target.
+ *
+ * @inflight `@winglang/sdk.std.ISharedStateClient`
+ */
+export abstract class SharedState extends Resource implements IResource {
```

Comment:
I am not sure the abstract version is required. Users should explicitly instantiate the concrete `sim.Object`.